docs/recipes/auth/README.md

```diff
@@ -0,0 +1,94 @@
+# Running Lighthouse on Authenticated Pages with Puppeteer
+
+If you just want to view the code, see [example-lh-auth.js](./example-lh-auth.js).
+
+## The Example Site
+
+There are two pages on the site:
+
+1. `/` - the homepage
+2. `/dashboard`
+
+The homepage shows the login form, but only to users that are not signed in.
+
+The dashboard shows a secret to users that are logged in, but shows an error to users that are not.
+
+The server responds with different HTML for each of these pages and session states, so there are four different pages that must have passable Lighthouse SEO scores.
+
+(Optional) To run the server:
+```sh
+# be in root lighthouse directory
+yarn # install global project deps
+cd docs/auth
+yarn # install deps related to just this documentation
+yarn start # start the server on http://localhost:8000
+```
+
+## Process
+
+Puppeteer - a browser automation tool - can be used to programatically setup a session.
+
+1. Launch a new browser.
+1. Navigate to the login page.
+1. Fill and submit the login form.
+1. Run Lighthouse using the same browser.
+
+First, launch Chrome:
+```js
+// This port will be used by Lighthouse later.
+const PORT = 8041;
+const browser = await puppeteer.launch({
+  args: [`--remote-debugging-port=${PORT}`],
+});
+```
+
+Navigate to the login form:
+```js
+const page = await browser.newPage();
+await page.goto('http://localhost:8000');
+```
+
+Given a login form like this:
+```html
+<form action="/login" method="post">
+  <label>
+    Email:
+    <input type="email" name="email">
+  </label>
+  <label>
+    Password:
+    <input type="password" name="password">
+  </label>
+  <input type="submit">
+</form>
+```
+
+Direct Puppeteer to fill and submit it:
+```js
+const emailInput = await page.$('input[type="email"]');
+await emailInput.type('admin@example.com');
+const passwordInput = await page.$('input[type="password"]');
+await passwordInput.type('password');
+const submitInput = await page.$('input[type="submit"]');
+await submitInput.press('Enter');
+await page.waitForNavigation();
+```
+
+At this point, the session that Puppeteer is managing is now logged in.
+
+Close the page used to login:
+```js
+await page.close();
+// The page has been closed, but the browser still has the relevant session.
+```
+
+Now run Lighthouse, using the same port as before:
+```js
+const result = await lighthouse('http://localhost:8000/dashboard', { port: PORT });
+await browser.close();
+const lhr = result.lhr;
+```
+
+## Puppetter in Your Integration Tests
+
+See [example-lh-auth.test.js](./example-lh-auth.test.js) for an example of how to run Lighthouse in your Jest tests on pages in both an authenticated and non-authenticated session.
```

Comment:
maybe move this to the top of the page alongside the `If you just want to view the code`? By the time I got down to the bottom through the other `example-lh-auth.js` example it wasn't clear this was a whole different thing.

And if I'm a new or repeat visitor looking specifically for an integration test example, it's helpful to have the link off at the top rather than deep in the page.

Reply:
your more recent comment suggests pulling the jest integration thing into its own recipe. I like and agree with that.

@connorjclark i'm thinking a recipes/integration-test folder. the readme would add in "btw: we're building on the [/auth recipe]". i think it makes sense to do this bit in a followup PR.